### PR TITLE
WebClient: Add multiple validator transactions

### PR DESCRIPTION
**Pull request by faberto**
Adding multiple validator related functions to the web-client

## What's in this pull request?
Add signal_data to `new_create_validator() ` 
Add Validator functions:
- `new_update_validator`
- `new_deactivate_validator`
- `new_delete_validator`
- `new_retire_validator`

## Pull request checklist

- [x] All tests pass. The project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have resolved all `clippy` and `rustfmt` warnings.

### CODE REVIEW

File: web-client/src/primitives/transaction_builder.rs [hunk 1]
```diff
@@ -267,16 +270,19 @@ impl TransactionBuilder {
         reward_address: &Address,
         signing_key: PublicKey,
         voting_key_pair: BLSKeyPair,
+        signal_data: Option<String>,
         fee: Option<u64>,
         validity_start_height: u32,
         network_id: u8,
     ) -> Result<Transaction, JsError> {
+        let native_signal_data: Option<Blake2bHash> =
+            signal_data.map_or(None, |r| Some(Blake2bHash::from_str(&r).unwrap()));
```
Comment (jsdanielh) [inline web-client/src/primitives/transaction_builder.rs]: Better to use `map` in this case

File: web-client/src/primitives/transaction_builder.rs [hunk 2]
```diff
@@ -267,16 +270,19 @@ impl TransactionBuilder {
...
         let mut recipient = Recipient::new_staking_builder();
         recipient.create_validator(
-            *signing_key.native_ref(),
+            signing_key.native_ref().clone(),
```
Comment (jsdanielh) [inline web-client/src/primitives/transaction_builder.rs]: This was actually correct as it was. Now it produces a clippy warning.

File: web-client/src/primitives/transaction_builder.rs [hunk 3]
```diff
@@ -293,11 +299,144 @@ impl TransactionBuilder {
...
+        reward_address: &Address, //Should also be optional Option<>
+        signing_key: Option<PublicKey>,
+        voting_key_pair: Option<BLSKeyPair>,
+        signal_data: Option<String>,
+        fee: Option<u64>,
+        validity_start_height: u32,
+        network_id: u8,
+    ) -> Result<Transaction, JsError> {
+        let native_signal_data: Option<Option<Blake2bHash>> =
+            signal_data.map_or(None, |r| Some(Some(Blake2bHash::from_str(&r).unwrap())));
```
Comment (jsdanielh) [inline web-client/src/primitives/transaction_builder.rs]: Better to use `map` instead

File: web-client/src/primitives/transaction_builder.rs [hunk 4]
```diff
@@ -293,11 +299,144 @@ impl TransactionBuilder {
...
+        let mut recipient = Recipient::new_staking_builder();
+        let native_signing_key: Option<nimiq_keys::PublicKey> =
+            signing_key.map_or(None, |r| Some(r.native_ref().clone()));
```
Comment (jsdanielh) [inline web-client/src/primitives/transaction_builder.rs]: Better to use `map` instead

File: web-client/src/primitives/transaction_builder.rs [hunk 5]
```diff
@@ -293,11 +299,144 @@ impl TransactionBuilder {
...
+        let native_voting_key_pair: Option<nimiq_bls::KeyPair> =
+            voting_key_pair.map_or(None, |r| Some(r.native_ref().clone()));
```
Comment (jsdanielh) [inline web-client/src/primitives/transaction_builder.rs]: Better to use `map` instead

File: web-client/src/primitives/transaction_builder.rs (hunk 4 above)
Comment (jsdanielh) [reply web-client/src/primitives/transaction_builder.rs]: `PublicKey` implements `Copy`, so better use: `*r.native_ref()`

File: web-client/src/primitives/transaction_builder.rs [hunk 6]
```diff
@@ -293,11 +299,144 @@ impl TransactionBuilder {
...
-    // pub fn new_update_validator()
+    /// Updates parameters of a validator in the staking contract.
+    ///
+    /// The returned transaction is not yet signed. You can sign it e.g. with `tx.sign(keyPair)`.
+    ///
+    /// Throws when the fee does not fit within a u64 or the `networkId` is unknown.
+    #[wasm_bindgen(js_name = newUpdateValidator)]
+    pub fn new_update_validator(
+        sender: &Address,
+        reward_address: &Address, //Should also be optional Option<>
```
Comment (jsdanielh) [inline web-client/src/primitives/transaction_builder.rs]: Why not `Option<Address>`?

**Review by jsdanielh — CHANGES_REQUESTED** [review]

File: web-client/src/primitives/transaction_builder.rs (hunk 6 above)
Comment (faberto) [reply web-client/src/primitives/transaction_builder.rs]: Updated. Is now also Optional

**Review by jsdanielh — COMMENTED** [review]
LGTM. Could you squash your commits and make sure the resulting commit is signed?

File: web-client/src/primitives/transaction_builder.rs [hunk 7]
```diff
@@ -293,11 +299,146 @@ impl TransactionBuilder {
...
-    // pub fn new_update_validator()
+    /// Updates parameters of a validator in the staking contract.
+    ///
+    /// The returned transaction is not yet signed. You can sign it e.g. with `tx.sign(keyPair)`.
+    ///
+    /// Throws when the fee does not fit within a u64 or the `networkId` is unknown.
+    #[wasm_bindgen(js_name = newUpdateValidator)]
+    pub fn new_update_validator(
+        sender: &Address,
+        reward_address: Option<Address>, //Should also be optional Option<>
```
Comment (jsdanielh) [inline web-client/src/primitives/transaction_builder.rs]:
```suggestion
        reward_address: Option<Address>,
```

**Review by jsdanielh — APPROVED** [review]

**Pull request merged**